src/EventListener/InitializeSystemListener.php

```diff
@@ -114,30 +51,14 @@ public function __construct(
      */
     public function onKernelRequest(GetResponseEvent $event)
     {
-        if (true === self::$booted || (!$this->isFrontendScope() && !$this->isBackendScope())) {
+        if (true === self::$booted || !$this->isContaoScope()) {
             return;
         }
 
         // Set before calling any methods to prevent recursive booting
         self::$booted = true;
 
-        $request = $event->getRequest();
-
-        $route = $this->router->generate(
-            $request->attributes->get('_route'),
-            $request->attributes->get('_route_params')
-        );
-
-        $basePath = $request->getBasePath();
-
-        $this->setConstants(
-            $this->getModeFromContainerScope(),
-            substr($route, strlen($basePath) + 1),
-            $basePath,
-            $request->attributes->get('_contao_referer_id')
-        );
-
-        $this->boot($request);
+        $this->eventDispatcher->dispatch('contao.framework.boot', new ContaoFrameworkBootEvent($event->getRequest()));
```

Comment:
Ok, I agree to have a consistence here. I would change this according to #204. 

BTW: Did you had a look at `RegisterListenerPass`. This maybe helpful for #204 
https://github.com/symfony/EventDispatcher/blob/master/DependencyInjection/RegisterListenersPass.php